### PR TITLE
Add concurrent interleave operations to Async module

### PR DESCRIPTION
**Pull request by harendra-kumar**

### CODE REVIEW

File: src/Streamly/Internal/Data/Stream/Async/Channel.hs [hunk 1]
```diff
@@ -35,19 +34,25 @@ module Streamly.Internal.Data.Stream.Async.Channel
     -- * Stream operations
     , evalWithK
     , appendWithK
-    , concatMapWithK
+    , interleaveWithK
+    , concatMapWithK -- appendMap
+    , concatMapInterleaveWithK -- interleaveMap
```
Comment (adithyaov) [inline src/Streamly/Internal/Data/Stream/Async/Channel.hs:39]: Append-ish
- `withChannelK` (Rename this to withAppendWithK. Basically indicate append-ish)
- `appendWithK`
- `evalWithK` (Rename this to indicate append-ish)
- `concatMapWithK` -- appendMap

Interleave-ish
- `withInterleaveChannelK`
- `interleaveWithK`
- `concatMapInterleaveWithK` -- interleaveMap

Comment (adithyaov) [reply src/Streamly/Internal/Data/Stream/Async/Channel.hs:39]: Or is the default append-ish.

Comment (adithyaov) [inline src/Streamly/Internal/Data/Stream/Async/Channel.hs:39]: If you're exporting `withChannelK`, Also export `withInterleaveChannelK`? Keeping the corresponding interleave function will help make the distinction clearer and in a sense describe if it's a general API or async strategy specific..

Comment (adithyaov) [reply src/Streamly/Internal/Data/Stream/Async/Channel.hs:39]: Also, `evalWithInterleaveK`.

Comment (harendra-kumar) [reply src/Streamly/Internal/Data/Stream/Async/Channel.hs:39]: Yes the default is append, to keep the default name shorter. I considered naming explicitly for both, but was not sure.

Comment (harendra-kumar) [reply src/Streamly/Internal/Data/Stream/Async/Channel.hs:39]: append or interleave does not apply to eval as it is a single stream. In fact, eval should only exist in the Stream.Parallel module.

**Pull request merged**